catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/plugins/FrameCenterUpdateFieldTest.java

```diff
@@ -0,0 +1,131 @@
+import static org.mockito.Mockito.when;
+import static org.mockito.internal.verification.VerificationModeFactory.times;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import org.codice.alliance.libs.klv.AttributeNameConstants;
+import org.codice.alliance.libs.klv.GeometryOperator;
+import org.codice.alliance.video.stream.mpegts.Context;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.io.ParseException;
+import com.vividsolutions.jts.io.WKTReader;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+
+public class FrameCenterUpdateFieldTest {
+
+    @Test
+    public void testLineString() {
+
+        String wktChild1 = "LINESTRING (30 10, 10 30, 40 40)";
+        String wktChild2 = "LINESTRING (50 50, 60 60, 70 70)";
+        String wktChild3 = "LINESTRING (80 80, 85 85)";
+
+        String wktExpected = "LINESTRING (30 10, 10 30, 40 40, 50 50, 60 60, 70 70, 80 80, 85 85)";
+
+        Metacard parentMetacard = mock(Metacard.class);
+
+        Metacard childMetacard1 = mock(Metacard.class);
+        when(childMetacard1.getAttribute(AttributeNameConstants.FRAME_CENTER)).thenReturn(new AttributeImpl(
+                AttributeNameConstants.FRAME_CENTER,
+                wktChild1));
+        Metacard childMetacard2 = mock(Metacard.class);
+        when(childMetacard2.getAttribute(AttributeNameConstants.FRAME_CENTER)).thenReturn(new AttributeImpl(
+                AttributeNameConstants.FRAME_CENTER,
+                wktChild2));
+        Metacard childMetacard3 = mock(Metacard.class);
+        when(childMetacard3.getAttribute(AttributeNameConstants.FRAME_CENTER)).thenReturn(new AttributeImpl(
+                AttributeNameConstants.FRAME_CENTER,
+                wktChild3));
+
+        FrameCenterUpdateField frameCenterUpdateField =
+                new FrameCenterUpdateField(GeometryOperator.IDENTITY, new GeometryFactory());
+
+        Context context = mock(Context.class);
+        GeometryOperator.Context geometryOperatorContext = new GeometryOperator.Context();
+        geometryOperatorContext.setSubsampleCount(1000);
+        when(context.getGeometryOperatorContext()).thenReturn(geometryOperatorContext);
+
+        frameCenterUpdateField.updateField(parentMetacard,
+                Collections.singletonList(childMetacard1),
+                context);
+        frameCenterUpdateField.updateField(parentMetacard,
+                Arrays.asList(childMetacard2, childMetacard3),
+                context);
+        frameCenterUpdateField.end(parentMetacard, context);
+
+        ArgumentCaptor<Attribute> captor = ArgumentCaptor.forClass(Attribute.class);
+
+        verify(parentMetacard).setAttribute(captor.capture());
+
+        assertThat(captor.getValue()
+                .getValue(), is(wktExpected));
```

Comment:
I think I understand what you mean, but please double check this class of unit tests after I push the new code.